Libs/MRML/Core/vtkMRMLSubjectHierarchyNode.cxx

```diff
@@ -1245,6 +1245,10 @@ void vtkSubjectHierarchyItem::ReparentChildrenToParent()
 //---------------------------------------------------------------------------
 void vtkSubjectHierarchyItem::RemoveAllChildren()
 {
+  // If the item has many children, then it may invoke a large number of events.
+  // To avoid this, block modified events on the data node until all of the children are removed.
```

Comment:
```suggestion
  // To avoid this, block Modified events on the data node until all of the children are removed.
  // Once all items have been removed, if it applies, a single Modified event is invoked.